sgxs-tools/src/sgx_detect/tests/mod.rs

```diff
@@ -784,8 +785,160 @@ impl Dependency<RunEnclaveProdWl> for RunEnclaveProd {
     const CONTROL_VISIBILITY: bool = true;
 }
 
+// EnclaveOS requirements
+#[derive(Debug, Default, DebugSupport)]
+struct Eor {
+    graphene: Status,
+    node_agent: Status,
+    perm_daemon: Status,
+    build_type: BuildType,
+}
+
+impl Update for Eor {
+    fn update(&mut self, support: &SgxSupport) {
+        self.graphene = match support.enclaveos_dev {
+                Ok(_) => Status::Supported,
+                Err(_) => Status::Fatal,
+            };
+        self.node_agent = match support.node_agent {
+                Ok(_) => Status::Supported,
+                Err(_) => Status::Fatal,
+            };
+        self.perm_daemon = match support.perm_daemon {
+                Ok(_) => Status::Supported,
+                Err(_) => Status::Fatal,
+            };
+        self.build_type = support.build_type;
+    }
+}
+
+impl Print for Eor {
+    fn supported(&self) -> Status {
+        if self.build_type == BuildType::EnclaveOSPreInstall {
+            self.graphene & self.perm_daemon
+        } else {
+            self.graphene & self.perm_daemon & self.node_agent
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, DebugSupport)]
+struct BuildConfig {
```

Comment:
The config or type doesn't really relate to the build anymore. Maybe a better term is `EnvConfig` or `EnvType`